### DISCUSSION TITLE
Authorization

### DISCUSSION THREAD

**Discussion by dawidow** [adonisjs/core#1804] [Help] [answered]
Hey folks. I came here with a question about authorization in AdonisJS v5. In Laravel we have Gate and @can tag in Blade files. How to resolve this in AdonisJS? For example, I wanna make Admin who can see extra links in menu like Create new post or something like that. Could you tell me how to do that? Thanks

**Comment by dawidow** [comment]
Any ideas? What is the best way for make Admin role and give him permissions to the path?

**Comment by Banashek** [comment, marked as answer]
There are a number of [packages](https://github.com/zgabievi/awesome-adonisjs#packages) that haven't yet been migrated over to v5, including [Adonis Guard](https://github.com/RomainLanz/adonis-guard) and [Adonis ACL](https://github.com/enniel/adonis-acl).

There are two options:

1. Roll your own authorization rules and helpers (can use the package as inspiration)
2. Upgrade one of the existing packages to v5 (probably mostly typescriptifying and adjusting IoC/Adonis package usage).

**Reply by dawidow** [reply]
Thanks for your answer. That’s the problem. I spend a lot of time searching for the solution before I’m writing this thread. I saw the Guard package is great but it’s still not supported in v5.

**Reply by Banashek** [reply]
Yeah v5 seems to be getting closer, but like any software things take time and a lot of these packages are written by other folks in the community who may-or-may-not still be actively maintaining them. You could always open an issue in the package asking either if there is a planned upgrade for v5 or if you want to help maybe ask if they'd like contributions.

You can always add  an `isAdmin` property to the [template state](https://preview.adonisjs.com/guides/views/data-flow#template-state) and use [a conditional](https://preview.adonisjs.com/guides/views/conditionals#ifelse) to render the template. This is a perfectly fine approach and one that most folks not using some view-integrated authorization package use all the time in production.

**Reply by dawidow** [reply]
@Banashek I found an interesting article. Do you think this is a good approach? And is it possible to achieve it in v5? In my case it would be interesting because I would like to do, for example, create a route create-post and only admin could enter there and only admin could see it in navigation.

[Link to article](https://github.com/adonisjs/auth/issues/121)

Here is an interesting example in the documentation with Acl in Route Middleware section - [Route middleware](https://preview.adonisjs.com/guides/http/middleware#route-middleware)

It looks very close to my needs.

**Reply by Banashek** [reply]
Yeah I don't see why not.
